File: src/org/cappuccino_project/ide/intellij/plugin/indices/ObjJClassInstanceVariableAccessorMethodIndex.java
package org.cappuccino_project.ide.intellij.plugin.indices;

import com.intellij.psi.stubs.StubIndexKey;
import org.cappuccino_project.ide.intellij.plugin.psi.ObjJInstanceVariableDeclaration;
import org.jetbrains.annotations.NotNull;

public class ObjJClassInstanceVariableAccessorMethodIndex extends ObjJStringStubIndexBase<ObjJInstanceVariableDeclaration> {

    private static final ObjJClassInstanceVariableAccessorMethodIndex INSTANCE = new ObjJClassInstanceVariableAccessorMethodIndex();
    private static final StubIndexKey<String, ObjJInstanceVariableDeclaration> KEY = IndexKeyUtil.createIndexKey(ObjJClassInstanceVariableAccessorMethodIndex.class);
    private static final int VERSION = 0;

    public static ObjJClassInstanceVariableAccessorMethodIndex getInstance() {
        return INSTANCE;
    }

    @NotNull
    @Override
    public StubIndexKey<String, ObjJInstanceVariableDeclaration> getKey() {
        return KEY;
    }

    @Override
    public int getVersion() {
        return super.getVersion() + VERSION;
    }

    @NotNull
    @Override
    protected Class<ObjJInstanceVariableDeclaration> getIndexedElementClass() {
        return ObjJInstanceVariableDeclaration.class;
    }
}

File: src/cappuccino/ide/intellij/plugin/indices/ObjJClassInstanceVariableAccessorMethodIndex.kt
package cappuccino.ide.intellij.plugin.indices

import cappuccino.ide.intellij.plugin.psi.ObjJInstanceVariableDeclaration
import com.intellij.psi.stubs.StubIndexKey

class ObjJClassInstanceVariableAccessorMethodIndex : ObjJStringStubIndexBase<ObjJInstanceVariableDeclaration>() {

    override val indexedElementClass: Class<ObjJInstanceVariableDeclaration>
        get() = ObjJInstanceVariableDeclaration::class.java

    override fun getKey(): StubIndexKey<String, ObjJInstanceVariableDeclaration> {
        return KEY
    }

    override fun getVersion(): Int {
        return super.getVersion() + VERSION
    }

    companion object {

        val instance = ObjJClassInstanceVariableAccessorMethodIndex()
        private val KEY = IndexKeyUtil.createIndexKey(ObjJClassInstanceVariableAccessorMethodIndex::class.java)
        private const val VERSION = 1
    }
}
